juju/juju
# refactor: migration target facade to a multi model context

This PR starts some cleanup needed on the migration target facade. Specifically removing versions 1 and 2 and also changing this facade over to a multi model facade.

We are removing version 1 and 2 as I would like to make this facade multi model and don't want to deal with moving  three version over for this support. Juju 3.6 supports version 3 of the facade so we can safely remove 1 and 2. The reason for moving this facade to a multi model context is we need to reach across models to perform operations in this facade. Specifically when having a model migrated into this controller the source controller migration master checks the final machines at the end from the perspective of the controller model reaching into the migrated model.

This work is being done so the credential validation context can be removed off of the facade allowing model config to get completed.

## Checklist


## QA steps

1. Bootstrap a source controller (3.6) and a target controller (4.0).
2. Add a new model to the source controller (foobar) and deploy ubuntu charm. Wait for everything to settle down.
3. Migrate the controller from the source to the target and confirm the operation is successful with no errors.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-6421
